sql/core/src/test/scala/org/apache/spark/sql/execution/RemoveRedundantSortsSuite.scala

```diff
@@ -124,6 +124,7 @@ abstract class RemoveRedundantSortsSuiteBase
   test("cached sorted data doesn't need to be re-sorted") {
     withSQLConf(SQLConf.REMOVE_REDUNDANT_SORTS_ENABLED.key -> "true") {
       val df = spark.range(1000).select($"id" as "key").sort($"key".desc).cache()
+      df.collect()
```

Comment:
why do we need to trigger table cache here?

Reply:
We propagate output partitioning and ordering when the inside AQE is `isFinalPlan`, so we should materialzed it first. For these tests, it's equal to add `resorted.collect()`.

Reply:
Got it, we need to trigger AQE execution to get the final plan if the tests check the physical plan.